packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js

```diff
@@ -95,27 +95,50 @@ const reducer = (state, action) =>
         const { value, replace = false } = action;
         const connectedRelations = get(state, [...path, 'connect']);
         const disconnectedRelations = get(state, [...path, 'disconnect']);
+        const savedRelations = get(state, [...path, 'results']);
```

Comment:
you're checking the existence of `disconnectRelations` and `savedRelations` a lot, you could do:

```ts
const savedRelations = get(state, [...path, 'results']) ?? [];
```

to save you some time and continuously checking if the array exists.

Reply:
I added it and I was able to remove the check on savedRelations in
```js
const existInSavedRelation =
          savedRelations?.findIndex((savedRelations) => savedRelations.id === value.id) !== -1;
``` 

most of the time I check on length though, didn't find another place where I could remove this type of check - did I miss something? 👀

Reply:
<img width="579" alt="CleanShot 2022-10-11 at 13 29 24@2x" src="https://user-images.githubusercontent.com/37798644/195090688-898e533d-1c99-41a7-bc5b-afc51952ae32.png">

all your `ifs` are `disconnectedRelations?.length` (essentially checking if the length is above 0) so by ensuring `disconnectedRelations` is _definitely_ an `array` via a `??` you can just just "disconnectedRelations.length` which is easier to understand and reduces the amount of optional chaining we do.